bundle/direct/dresources/secret_scope_acls.go

```diff
@@ -0,0 +1,223 @@
+package dresources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"slices"
+	"strings"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/bundle/deployplan"
+	"github.com/databricks/cli/libs/structs/structvar"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/workspace"
+	"golang.org/x/sync/errgroup"
+)
+
+type ResourceSecretScopeAcls struct {
+	client *databricks.WorkspaceClient
+}
+
+type SecretScopeAclsState struct {
+	ScopeName string              `json:"scope_name"`
+	Acls      []workspace.AclItem `json:"acls,omitempty"`
+}
+
+func PrepareSecretScopeAclsInputConfig(inputConfig any, node string) (*structvar.StructVar, error) {
+	baseNode, ok := strings.CutSuffix(node, ".permissions")
+	if !ok {
+		return nil, fmt.Errorf("internal error: node %q does not end with .permissions", node)
+	}
+
+	// Use reflection to get the slice from the pointer
+	rv := reflect.ValueOf(inputConfig)
+	if rv.Kind() != reflect.Ptr || rv.Elem().Kind() != reflect.Slice {
+		return nil, fmt.Errorf("inputConfig must be a pointer to a slice, got: %T", inputConfig)
+	}
+
+	sliceValue := rv.Elem()
+
+	// Convert slice to []resources.SecretScopePermission
+	acls := make([]workspace.AclItem, 0, sliceValue.Len())
+	for i := range sliceValue.Len() {
+		elem := sliceValue.Index(i).Interface().(resources.SecretScopePermission)
+		acl := workspace.AclItem{
+			Permission: workspace.AclPermission(elem.Level),
+			Principal:  "",
+		}
+		if elem.UserName != "" {
+			acl.Principal = elem.UserName
+		} else if elem.GroupName != "" {
+			acl.Principal = elem.GroupName
+		} else if elem.ServicePrincipalName != "" {
+			acl.Principal = elem.ServicePrincipalName
+		}
+		acls = append(acls, acl)
+	}
+
+	// Sort ACLs by principal for deterministic ordering
+	slices.SortFunc(acls, func(a, b workspace.AclItem) int {
+		return strings.Compare(a.Principal, b.Principal)
+	})
+
+	return &structvar.StructVar{
+		Value: &SecretScopeAclsState{
+			ScopeName: "", // Always a reference, defined in Refs below
+			Acls:      acls,
+		},
+		Refs: map[string]string{
+			"scope_name": "${" + baseNode + ".name}",
+		},
+	}, nil
+}
+
+func (*ResourceSecretScopeAcls) New(client *databricks.WorkspaceClient) *ResourceSecretScopeAcls {
+	return &ResourceSecretScopeAcls{client: client}
+}
+
+func (*ResourceSecretScopeAcls) PrepareState(s *SecretScopeAclsState) *SecretScopeAclsState {
+	return s
+}
+
+func (r *ResourceSecretScopeAcls) DoRead(ctx context.Context, id string) (*SecretScopeAclsState, error) {
+	// id is the scope name
+	currentAcls, err := r.client.Secrets.ListAclsAll(ctx, workspace.ListAclsRequest{
+		Scope: id,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list ACLs: %w", err)
```

Comment:
Wrapping SDK errors is not necessary here, we don't do it in other resources.

I suggest to add acceptance test with endpoint returning an error, to see how it looks.